### PR TITLE
Dockerfile and basic CI for Go relay maintainer

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Depends on:  #762

This pull request introduces the `Dockerfile` for the Go relay maintainer along with a basic `build-test-publish` CI job.

### CODE REVIEW

File: .github/workflows/relay.yml
Comment (michalinacienciala) [inline .github/workflows/relay.yml:58]: After [this discussion in Flowdock](https://www.flowdock.com/app/cardforcoin/tech/threads/sqQIWXEcGU8O2Gf-mMqgsfmT6wr) I updated `keep-core` and `keep-ecdsa` workflows to execute steps publishing client only when workflow runs due to merge to master. Changes of the config aren't yet in the master, they're waiting for review in https://github.com/keep-network/keep-core/pull/2378 and https://github.com/keep-network/keep-ecdsa/pull/738. Do we want same rules for publishing of relay maintainer (that wasn't discussed in FD)? If yes, we can apply same config here.
```suggestion
          github.ref == 'refs/heads/master'
            && (github.event_name == 'push'
            || github.event_name == 'workflow_dispatch')
```

File: .github/workflows/relay.yml
Comment (michalinacienciala) [inline .github/workflows/relay.yml:80]: Same thing as with the comment for line 58:
```suggestion
          push: |
            ${{ github.ref == 'refs/heads/master'
              && (github.event_name == 'push'
              || github.event_name == 'workflow_dispatch') }}
```

File: .github/workflows/relay.yml
Comment (michalinacienciala) [inline .github/workflows/relay.yml:38]: We no longer need to monitor the disk space, so those line can be removed (I removed them in `keep-core` and `keep-ecdsa` in recent PRs - not yet merged).

**Review by michalinacienciala — COMMENTED** [review]
I left some comments regarding ci job. I also took a look at Dockerfile, but I'm not an expert on it, so I'm leaving it for review for somebody more experienced.

File: .github/workflows/relay.yml
Comment (lukasz-zimnoch) [reply .github/workflows/relay.yml:58]: Yep I think it's a good idea. Please see https://github.com/keep-network/tbtc/commit/724110da01c83b57569db5d7120132bd27732ce0

File: .github/workflows/relay.yml
Comment (lukasz-zimnoch) [reply .github/workflows/relay.yml:80]: Please see https://github.com/keep-network/tbtc/commit/724110da01c83b57569db5d7120132bd27732ce0

File: .github/workflows/relay.yml
Comment (lukasz-zimnoch) [reply .github/workflows/relay.yml:38]: Please see https://github.com/keep-network/tbtc/commit/724110da01c83b57569db5d7120132bd27732ce0

**Comment by lukasz-zimnoch** [comment]
@michalinacienciala ready for another look!

**Review by michalinacienciala — APPROVED** [review]

**Review by tomaszslabon — APPROVED** [review]

**Pull request merged**
